### PR TITLE
Filter files before SOPS decryption

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:

* Decrypt YAMLs with SOPS only if the file has a 'sops' field. This should speed up the apply time for large repositories. 
* Update SOPS package and binary to v3.6.0

### CODE REVIEW

File: pkg/cluster/kubernetes/resource/load.go [hunk 1]
```diff
@@ -207,7 +208,7 @@ func loadFile(path string, sopsEnabled bool) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	if sopsEnabled {
+	if sopsEnabled && strings.Contains(string(bytes), "sops:") {
```
Comment (squaremo) [inline pkg/cluster/kubernetes/resource/load.go:211]: `bytes.Contains(bytes, []byte("sops:"))` (or better still, make that latter a var)

Comment (squaremo) [reply pkg/cluster/kubernetes/resource/load.go:211]: (you'll have to rename the variable of course)

File: pkg/cluster/kubernetes/resource/load.go [hunk 2]
```diff
@@ -203,14 +203,14 @@ func ParseMultidoc(multidoc []byte, source string) (map[string]KubeManifest, err
...
 // it will try to decrypt it before returning the data
 func loadFile(path string, sopsEnabled bool) ([]byte, error) {
-	bytes, err := ioutil.ReadFile(path)
+	fileBytes, err := ioutil.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
-	if sopsEnabled {
-		return softDecrypt(bytes)
+	if sopsEnabled && bytes.Contains(fileBytes, []byte("sops:")) {
```
Comment (redbaron) [inline pkg/cluster/kubernetes/resource/load.go:210]: unlikely to be triggered IRL, but this disables sops for JSON-like content of yaml files

**Pull request merged**